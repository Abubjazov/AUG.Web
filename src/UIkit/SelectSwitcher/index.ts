import SelectSwitcher from './SelectSwitcher'

export default SelectSwitcher
