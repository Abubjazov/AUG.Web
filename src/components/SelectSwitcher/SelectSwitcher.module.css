.root {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 270px;
  height: 40px;
  padding: 0 15px;
  background: var(--white);
  border: none;
  border-radius: 10px;
  box-shadow: var(--box-shadow);
  animation: var(--animation-visible);
}

.span-wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  cursor: pointer;
}

.root span {
  display: flex;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 10px 15px;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  color: var(--logotext);
  cursor: pointer;
  background: none;
  background-image: none;
}

.option:hover {
  background: var(--gray);
}

.drop-down {
  position: absolute;
  top: 43px;
  right: 0;
  z-index: 100;
  display: flex;
  flex-direction: column;
  gap: 10px;
  width: 100%;
  background: var(--white);
  border: none;
  border-radius: 10px;
  box-shadow: var(--box-shadow);
  animation: var(--animation-visible);
}

.drop-down span:hover {
  color: var(--white);
}
