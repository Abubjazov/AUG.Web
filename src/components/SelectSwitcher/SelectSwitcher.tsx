import { FC, useEffect, useState, MouseEvent } from 'react'

import { useAppDispatch, useAppSelector } from 'store/hooks'
import { EOrderBy, orderDapplets, setOrderBy } from 'store/slices/dappletsSlice'
import SvgIcon from 'uikit/SvgIcon'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './SelectSwitcher.module.css'

export interface SelectSwitcherProps {
  userStyles?: string
}

const SelectSwitcher: FC<SelectSwitcherProps> = ({ userStyles = '' }) => {
  const { orderBy } = useAppSelector(state => state.dapplets)

  const [isOpenedDropDown, setIsOpenedDropDown] = useState(false)

  const dispatch = useAppDispatch()

  useEffect(() => {
    dispatch(orderDapplets())
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [orderBy])

  const openCloseDropDown = () => setIsOpenedDropDown(!isOpenedDropDown)

  const changeOrderBy = (
    e: MouseEvent<HTMLSpanElement, globalThis.MouseEvent>,
  ) => {
    dispatch(setOrderBy(e.currentTarget.innerText as EOrderBy))
    openCloseDropDown()
  }

  return (
    <div className={cc([styles.root, userStyles])}>
      <div className={styles['span-wrapper']} onClick={openCloseDropDown}>
        <span
          style={orderBy ? { padding: 0 } : { padding: 0, color: '#7F7F7F' }}
        >
          {orderBy ? orderBy : 'orderBy'}
        </span>

        <SvgIcon icon={'arrowDown'} />
      </div>

      {isOpenedDropDown && (
        <div className={styles['drop-down']} onMouseLeave={openCloseDropDown}>
          <div
            className={styles['option']}
            style={{
              borderTopRightRadius: '10px',
              borderTopLeftRadius: '10px',
            }}
          >
            <span
              style={
                orderBy === EOrderBy.ASC_BY_NAME
                  ? {
                      color: '#7F7F7F',
                    }
                  : {}
              }
              onClick={changeOrderBy}
            >
              {EOrderBy.ASC_BY_NAME}
            </span>
          </div>

          <span
            className={styles['option']}
            style={
              orderBy === EOrderBy.DESC_BY_NAME ? { color: '#7F7F7F' } : {}
            }
            onClick={changeOrderBy}
          >
            {EOrderBy.DESC_BY_NAME}
          </span>

          <span
            className={styles['option']}
            style={
              orderBy === EOrderBy.ASC_BY_MARKET_CAP ? { color: '#7F7F7F' } : {}
            }
            onClick={changeOrderBy}
          >
            {EOrderBy.ASC_BY_MARKET_CAP}
          </span>

          <div
            className={styles['option']}
            style={{
              borderBottomRightRadius: '10px',
              borderBottomLeftRadius: '10px',
            }}
            onClick={changeOrderBy}
          >
            <span
              style={
                orderBy === EOrderBy.DESC_BY_MARKET_CAP
                  ? { color: '#7F7F7F' }
                  : {}
              }
              onClick={changeOrderBy}
            >
              {EOrderBy.DESC_BY_MARKET_CAP}
            </span>
          </div>
        </div>
      )}
    </div>
  )
}

export default SelectSwitcher
