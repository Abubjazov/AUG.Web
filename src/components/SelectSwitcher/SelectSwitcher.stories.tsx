import type { Meta, StoryObj } from '@storybook/react'
import MockedProvider from 'mockData/mockedReduxProvider'

import SelectSwitcher, { SelectSwitcherProps } from './SelectSwitcher'

import '/src/index.css'

const meta: Meta<SelectSwitcherProps> = {
  component: SelectSwitcher,
  title: 'COMPONENTS/SelectSwitcher',
  parameters: {
    docs: {
      description: {
        component:
          'Interface component - SelectSwitcher. Used to configure content sorting',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
  },
}

export default meta

type Story = StoryObj<SelectSwitcherProps>

export const Default: Story = {
  args: {},
  decorators: [
    Story => (
      <MockedProvider>
        <div style={{ height: '230px' }}>
          <Story />
        </div>
      </MockedProvider>
    ),
  ],
}
