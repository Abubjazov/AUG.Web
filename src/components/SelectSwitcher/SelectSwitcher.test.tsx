import { screen, fireEvent, render } from '@testing-library/react'
import MockedProvider from 'mockData/mockedReduxProvider'
import { EOrderBy } from 'store/slices/dappletsSlice'
import * as actions from 'store/slices/dappletsSlice'

import SelectSwitcher from './SelectSwitcher'

describe('SelectSwitcher', () => {
  test('should render SelectSwitcher default', () => {
    const { asFragment } = render(
      <MockedProvider>
        <SelectSwitcher />
      </MockedProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render SelectSwitcher "Dropdown opened"', () => {
    const { asFragment } = render(
      <MockedProvider>
        <SelectSwitcher />
      </MockedProvider>,
    )

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_NAME"', () => {
    const mockedSetOrderBy = vi.spyOn(actions, 'setOrderBy')

    const { asFragment } = render(
      <MockedProvider>
        <SelectSwitcher />
      </MockedProvider>,
    )

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByText(EOrderBy.ASC_BY_NAME))

    expect(asFragment()).toMatchSnapshot()

    expect(mockedSetOrderBy).toHaveBeenCalledTimes(1)
    expect(mockedSetOrderBy).toHaveBeenCalledWith(EOrderBy.ASC_BY_NAME)

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_MARKET_CAP"', () => {
    const mockedSetOrderBy = vi.spyOn(actions, 'setOrderBy')

    const { asFragment } = render(
      <MockedProvider>
        <SelectSwitcher />
      </MockedProvider>,
    )

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByText(EOrderBy.DESC_BY_MARKET_CAP))

    expect(asFragment()).toMatchSnapshot()

    expect(mockedSetOrderBy).toHaveBeenCalledTimes(1)
    expect(mockedSetOrderBy).toHaveBeenCalledWith(EOrderBy.DESC_BY_MARKET_CAP)

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_NAME"', () => {
    const mockedSetOrderBy = vi.spyOn(actions, 'setOrderBy')

    const { asFragment } = render(
      <MockedProvider>
        <SelectSwitcher />
      </MockedProvider>,
    )

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByText(EOrderBy.DESC_BY_NAME))

    expect(asFragment()).toMatchSnapshot()

    expect(mockedSetOrderBy).toHaveBeenCalledTimes(1)
    expect(mockedSetOrderBy).toHaveBeenCalledWith(EOrderBy.DESC_BY_NAME)

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_MARKET_CAP"', () => {
    const mockedSetOrderBy = vi.spyOn(actions, 'setOrderBy')

    const { asFragment } = render(
      <MockedProvider>
        <SelectSwitcher />
      </MockedProvider>,
    )

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByText(EOrderBy.ASC_BY_MARKET_CAP))

    expect(asFragment()).toMatchSnapshot()

    expect(mockedSetOrderBy).toHaveBeenCalledTimes(1)
    expect(mockedSetOrderBy).toHaveBeenCalledWith(EOrderBy.ASC_BY_MARKET_CAP)

    fireEvent.click(screen.getByTestId('open-dropdown'))

    expect(asFragment()).toMatchSnapshot()
  })
})
