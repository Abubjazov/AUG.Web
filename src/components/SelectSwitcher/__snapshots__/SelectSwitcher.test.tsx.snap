// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_MARKET_CAP" 1`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        DESC by name
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span>
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
        style="color: rgb(127, 127, 127);"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span>
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_MARKET_CAP" 2`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        ASC by market cap
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_MARKET_CAP" 3`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        ASC by market cap
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span>
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
        style="color: rgb(127, 127, 127);"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span>
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_NAME" 1`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px; color: rgb(127, 127, 127);"
      >
        orderBy
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span>
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span>
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_NAME" 2`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        ASC by name
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.ASC_BY_NAME" 3`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        ASC by name
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span
          style="color: rgb(127, 127, 127);"
        >
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span>
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_MARKET_CAP" 1`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        ASC by name
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span
          style="color: rgb(127, 127, 127);"
        >
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span>
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_MARKET_CAP" 2`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        DESC by market cap
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_MARKET_CAP" 3`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        DESC by market cap
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span>
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span
          style="color: rgb(127, 127, 127);"
        >
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_NAME" 1`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        DESC by market cap
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span>
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span
          style="color: rgb(127, 127, 127);"
        >
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_NAME" 2`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        DESC by name
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" & switched to "EOrderBy.DESC_BY_NAME" 3`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px;"
      >
        DESC by name
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span>
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
        style="color: rgb(127, 127, 127);"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span>
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" 1`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px; color: rgb(127, 127, 127);"
      >
        orderBy
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher "Dropdown opened" 2`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px; color: rgb(127, 127, 127);"
      >
        orderBy
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="_drop-down_e3e346"
    >
      <div
        class="_option_e3e346"
        style="border-top-right-radius: 10px; border-top-left-radius: 10px;"
      >
        <span>
          ASC by name
        </span>
      </div>
      <span
        class="_option_e3e346"
      >
        DESC by name
      </span>
      <span
        class="_option_e3e346"
      >
        ASC by market cap
      </span>
      <div
        class="_option_e3e346"
        style="border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
      >
        <span>
          DESC by market cap
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectSwitcher > should render SelectSwitcher default 1`] = `
<DocumentFragment>
  <div
    class="_root_e3e346 "
  >
    <div
      class="_span-wrapper_e3e346"
      data-testid="open-dropdown"
    >
      <span
        style="padding: 0px; color: rgb(127, 127, 127);"
      >
        orderBy
      </span>
      <div>
        <svg
          height="10px"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 128.000000 128.000000"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#7F7F7F"
            stroke="none"
            transform="translate(0.000000,128.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M32 980 c-29 -27 -37 -64 -23 -99 7 -15 142 -157 300 -314 l287 -287 44 0 44 0 287 287 c158 157 293 299 300 314 24 60 -17 119 -85 119 -38 0 -47 -7 -293 -252 l-253 -252 -253 252 c-246 245 -255 252 -293 252 -27 0 -48 -7 -62 -20z"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
